src/TwigComponent/tests/Unit/TwigPreLexerTest.php

```diff
@@ -183,6 +183,45 @@ public static function getLexTests(): iterable
             EOF
         ];
 
+        yield 'component_where_entire_default_block_is_twig_embed' => [
+            <<<EOF
+            <twig:Alert>
+                <p>
+                    {% embed "my_embed.html.twig" with { foo: 'bar' } %}{% endembed %}
+                </p>
+            </twig:Alert>
+            EOF,
+            <<<EOF
+            {% component 'Alert' %}
+                {% block content %}<p>
+                    {% embed "my_embed.html.twig" with { foo: 'bar' } %}{% endembed %}
+                </p>
+            {% endblock %}{% endcomponent %}
+            EOF,
+        ];
+        yield 'component_where_entire_default_block_is_twig_embed_with_blocks' => [
+            <<<EOF
+            <twig:Alert>
+                <p>
+                    {% embed "my_embed.html.twig" %}
+                        {% block my_embed_block_1 "foo" %}
```

Comment:
I'd love to see this tested independently, both with a string - like '"foo"` - and also a variable like `{% block my_block fooVar %}`